MetaMask/metamask-extension
# fix: Permit value decimal & refactor: Create reusable ConfirmInfoRowTextTokenUnits component

## **Description**

No UI/UX changes

Create reusable ConfirmInfoRowTextTokenUnits

This also fixes the missing token decimals we had on the Permit value

notes: 
- there is an outstanding bug ticket to redesign the display length of the amount to max 15 characters. This will be handled separately
- there's also a precision issue being handle separately

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/26105?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/2888

## **Manual testing steps**

1. Create a redesign Permit transaction
2. Observe amount displays as expected

## **Screenshots/Recordings**



### **Before**



### **After**

![CleanShot 2024-07-25 at 03 35 08](https://github.com/user-attachments/assets/92e6db4e-7618-43d6-8259-d9e16f436dbe)


## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
